src/biogeochem/Fan2CTSMMod.F90

```diff
@@ -0,0 +1,1059 @@
+       nh3_flux_stores, nh3_flux_barns, man_n_transf, &
+       grz_fract, man_n_barns, tan_fract_excr, &
+       filter_soilc, num_soilc)
+    !
+    ! Evaluate storage losse for manure in the mixed/landless production systems
+    ! associated with crop columns in CLM. The N remaining after storage losses is applied
+    ! on soil either in the same column, or a fraction may be moved to the native
+    ! vegetation column within the gridcell (grasslands). This subroutine also evaluates
+    ! seasonal grazing of livestock, and the manure N on pastures is also moved into the
+    ! native vegatation.
+    ! 
+    use landunit_varcon, only : max_lunit
+    use pftconMod, only : nc4_grass, nc3_nonarctic_grass, nc3_arctic_grass
+    use clm_varcon, only : ispval
+    use landunit_varcon,      only:  istsoil, istcrop
+    use abortutils     , only : endrun
+    use LandunitType   , only: lun
+    use GridcellType   , only: grc
+    use clm_varctl     , only : iulog
+
+    implicit none
+    type(bounds_type), intent(in)    :: bounds
+    type(temperature_type) , intent(in) :: temperature_inst
+    type(frictionvel_type) , intent(in) :: frictionvel_inst
+    real(r8), intent(in) :: dt
+    
+    ! N excreted in manure, gN/m2:
+    real(r8), intent(in) :: ndep_sgrz_grc(bounds%begg:bounds%endg) ! seasonally grazing animals
+    real(r8), intent(in) :: ndep_ngrz_grc(bounds%begg:bounds%endg) ! non-grazing animals
+    ! N, TAN spread on grasslands, gN/m2/s:
+    real(r8), intent(inout) :: n_manure_spread_col(bounds%begc:bounds%endc) 
+    real(r8), intent(out) :: tan_manure_spread_col(bounds%begc:bounds%endc) ! output, calculated from the above and stored manure
+    ! N excreted by animals allocated to mixed production systems temporarily grazing on grasslands:
+    real(r8), intent(inout) :: n_manure_graze_col(bounds%begc:bounds%endc)
+    ! N excreted by animals in mixed systems, total
+    real(r8), intent(out) :: n_manure_mixed_col(bounds%begc:bounds%endc)
+    ! NH3 emission fluxes from manure storage and housings, gN/m2/s
+    real(r8), intent(out) :: nh3_flux_stores(bounds%begc:bounds%endc), nh3_flux_barns(bounds%begc:bounds%endc)
+    ! total nitrogen flux transferred out of a crop column (manure spreading + temporary grazing)
+    real(r8), intent(out) :: man_n_transf(bounds%begc:bounds%endc)
+    ! Total nitrogen excreted in barns
+    real(r8), intent(out) :: man_n_barns(bounds%begc:bounds%endc)
+    ! fraction of manure excreted when grazing
+    real(r8), intent(out) :: grz_fract(bounds%begc:bounds%endc)
+    ! TAN fraction in excreted N
+    real(r8), intent(in) :: tan_fract_excr
+    integer                  , intent(in)    :: num_soilc       ! number of soil columns in filter
+    integer                  , intent(in)    :: filter_soilc(:) ! filter for soil columns
+
+    integer :: begg, endg, g, l, c, il, counter, col_grass, status, p
+    real(r8) :: flux_avail_rum, flux_avail_mg, flux_grazing, invscale
```

Comment:
Here and elsewhere: the standard in CTSM is to have one variable declaration per line, with documentation of the variable that includes units.